src/screens/Root.js

```diff
@@ -1,17 +1,28 @@
 // @flow
 import React from 'react';
-import { Route, Switch } from 'react-router-dom';
+import { Route, Switch, Redirect } from 'react-router-dom';
 
 import HomeContainer from 'containers/HomeContainer/index';
 import Login from './Login/index';
 import OnboardingScreen from './OnboardingScreen';
 import WelcomeScreen from './WelcomeScreen';
 
-export default () => (
-  <Switch>
-    <Route exact path="/" component={HomeContainer} />
-    <Route exact path="/login" component={Login} />
-    <Route exact path="/onboarding" component={WelcomeScreen} />
-    <Route path="/onboarding/:type" component={OnboardingScreen} />
-  </Switch>
-);
+export default () => {
+  if (!localStorage.getItem('chainName')) {
+    return (
+      <Switch>
+        <Route exact path="/onboarding" component={WelcomeScreen} />
```

Comment:
We need to be careful in the order in which the `children` to [`<Switch />`](https://reacttraining.com/react-router/web/api/Switch) appear.

This is just a side-note. I’m not sure if you’re already aware, (since you’ve done it properly in this case, but I just want to be thorough) but RR4 will try and load each route, and keep trickling down the `children` array to find the most suitable route, failing which — I think — RR4 will redirect the user to a 404.

So as you can see, it now matters how we order our `<Route />` components.

On that note, super stoked you got this logic in place already!

Reply:
sounds good, and yeah we could make modifications to it later to show a 404 page when necessary